source/code/source/playfab/QoS/PlayFabQoSApi.cpp

```diff
@@ -59,8 +59,8 @@ namespace PlayFab
 
         void WriteTelemetryEvents(
             WriteEventsRequest& request,
-            ProcessApiCallback<WriteEventsResponse> callback,
-            ErrorCallback errorCallback = nullptr,
+            const ProcessApiCallback<WriteEventsResponse>& callback,
+            const ErrorCallback& errorCallback = nullptr,
```

Comment:
I'm not sure I agree with these...
A pointer is just an int64? There's no real reason to accept these by ref... Except these are std::func...
Ok, please explain the full context here. I think I may be vaguely understanding the point here, but please explain it for me, so I can fully grok.

Reply:
Mostly, I'm concerned about what the caller might do with their version of the std::func, and whether it's safe to keep the obj ref, or safer to copy.

Reply:
So this change is one I was also a bit on the fence with.

a callback should just be a function pointer, which const & wouldn't do much for

std::function on the other hand has some header info that would come with a more object like param.

From our perspective we will never change that function. BUT would that mean they couldn't do something like pass a lambda with this? (From what I'm seeing in our tests, this solution does still work).

Reply:
That's a valid concern.. References can be extremely dangerous when we don't control the lifespan of the argument

---
In reply to: [337640191](https://github.com/PlayFab/XPlatCoreTemplate/pull/24#discussion_r337640191) [](ancestors = 337640191)